_delphi_utils_python/delphi_utils/validator/datafetcher.py

```diff
@@ -158,10 +196,30 @@ def fetch_api_reference(data_source, start_date, end_date, geo_type, signal_type
 
     Formatting is changed to match that of source data CSVs.
     """
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        api_df = covidcast.signal(
-            data_source, signal_type, start_date, end_date, geo_type)
+    if start_date > end_date:
+        raise ValueError(
+            "end_day must be on or after start_day, but " f"start_day = '{start_date}', end_day = '{end_date}'"
+        )
+    response = Epidata.covidcast(
+        data_source,
+        signal_type,
+        time_type="day",
+        geo_type=geo_type,
+        time_values=Epidata.range(start_date.strftime("%Y%m%d"), end_date.strftime("%Y%m%d")),
+        geo_value=geo_type,
+    )
+    if response["result"] != 1:
+        # Something failed in the API and we did not get real metadata
+        raise RuntimeError("Error when fetching signal data from the API", response["message"])
+
+    api_df = None
+    if len(response["epidata"]) > 0:
+        api_df = pd.DataFrame.from_dict(response["epidata"])
+        api_df["issue"] = pd.to_datetime(api_df["issue"], format="%Y%m%d")
+        api_df["time_value"] = pd.to_datetime(api_df["time_value"], format="%Y%m%d")
```

Comment:
You said validator only runs on daily signals right? Because this will fail on signals where `time_value` is in epiweek `YYYYWW` format. Should be fine if we don't need to handle those now. Can be adjusted when we expand later.

Reply:
Ah good catch! I'll make a comment for future reference to fix them